lib/zod/schema.ts

```diff
@@ -1,77 +1,184 @@
+  name: teamName,
 });
 
 export const deleteApiKeySchema = z.object({
   apiKeyId: z.string(),
 });
 
 export const teamSlugSchema = z.object({
-  slug: z.string().max(maxLengthPolicies.slug),
+  slug,
 });
 
 export const updateTeamSchema = z.object({
-  name: z.string().min(1, 'Name is required').max(maxLengthPolicies.team),
-  slug: z
-    .string()
-    .min(3, 'Slug must be at least 3 characters')
-    .max(maxLengthPolicies.slug)
-    .transform((slug) => slugify(slug)),
-  domain: z
-    .string()
-    .max(maxLengthPolicies.domain)
-    .optional()
-    .refine(
-      (domain) => {
-        if (!domain) {
-          return true;
-        }
-
-        return isValidDomain(domain);
-      },
-      {
-        message: 'Enter a domain name in the format example.com',
-      }
-    )
-    .transform((domain) => {
-      if (!domain) {
-        return null;
-      }
-
-      return domain.trim().toLowerCase();
-    }),
+  name: teamName,
+  slug: slug.transform((slug) => slugify(slug)),
+  domain,
 });
 
 export const createTeamSchema = z.object({
-  name: z.string().min(1, 'Name is required').max(maxLengthPolicies.team),
+  name: teamName,
 });
 
 export const updateAccountSchema = z.union([
   z.object({
-    email: z
-      .string()
-      .email('Enter a valid email address')
-      .max(maxLengthPolicies.email),
+    email,
+  }),
+  z.object({
+    name,
+  }),
+  z.object({
+    image,
+  }),
+]);
+
+export const updatePasswordSchema = z.object({
+  currentPassword: password,
+  newPassword: password,
+});
+
+export const userJoinSchema = z.union([
+  z.object({
+    team: teamName,
+    slug,
   }),
   z.object({
-    name: z.string().min(1, 'Name is required').max(maxLengthPolicies.name),
+    name,
+    email,
+    password,
+  }),
+]);
+
+export const resetPasswordSchema = z.object({
+  password,
+  token: z.string(),
+});
+
+export const inviteViaEmailSchema = z.union([
+  z.object({
+    email,
+    role: z.nativeEnum(Role),
   }),
   z.object({
-    image: z
+    role: z.nativeEnum(Role),
+    domains: z
       .string()
-      .url('Enter a valid URL')
       .refine(
-        (imageUri) => imageUri.startsWith('data:image/'),
-        'Avatar must be an image'
-      )
-      .refine((imageUri) => {
-        const [, base64] = imageUri.split(',');
-        const size = base64.length * (3 / 4) - 2;
-        return size < 2000000;
-      }, 'Avatar must be less than 2MB'),
+        (domains) => domains.split(',').every(isValidDomain),
+        'Invalid domain in the list'
+      ),
   }),
 ]);
+
+export const resendLinkRequestSchema = z.object({
+  email,
+  expiredToken: z.string(),
+});
+
+export const deleteSessionSchema = z.object({
+  id: z.string(),
+});
+
+export const forgotPasswordSchema = z.object({
+  email,
+});
+
+export const resendEmailToken = z.object({
+  email,
+});
+
+export const checkoutSessionSchema = z.object({
+  priceId: z.string(),
+  quantity: z.number().optional(),
+});
```

Comment:
@ukrocks007 Be careful using the `z.string()`, an empty string is also a valid string. [More](https://github.com/colinhacks/zod/issues/2466)

Reply:
My point is

```
validateWithSchema(deleteSessionSchema, {}); // Fails
validateWithSchema(deleteSessionSchema, {id: ""}); // Succeeds
```

Reply:
yes looks like i need to add `.min(1)`